### PR TITLE
Timeseries updates

### PR DESCRIPTION
**Pull request by ehsteve**
This is work in progress

### CODE REVIEW

File: sunpy/timeseries/sources/goes.py [hunk 1]
```diff
@@ -192,10 +191,9 @@ def _parse_hdus(cls, hdulist):
         data = DataFrame({'xrsa': newxrsa, 'xrsb': newxrsb}, index=times)
         data.sort_index(inplace=True)
 
-        # Add the units data
-        units = OrderedDict([('xrsa', u.ct),
-                                ('xrsb', u.ct)])
-        # ToDo: check: http://ngdc.noaa.gov/stp/satellite/goes/doc/GOES_XRS_readme.pdf
+        # Add the units
+        units = OrderedDict([('xrsa', u.W/u.m**2),
+                                ('xrsb', u.W/u.m**2)])
```
Comment (Cadair) [inline sunpy/timeseries/sources/goes.py]: indentation

File: sunpy/timeseries/sources/rhessi.py [hunk 2]
```diff
@@ -123,15 +123,15 @@ def _parse_hdus(cls, hdulist):
...
-        units = OrderedDict([('3 - 6 keV', u.dimensionless_unscaled),
-                             ('6 - 12 keV', u.dimensionless_unscaled),
-                             ('12 - 25 keV', u.dimensionless_unscaled),
-                             ('25 - 50 keV', u.dimensionless_unscaled),
-                             ('50 - 100 keV', u.dimensionless_unscaled),
-                             ('100 - 300 keV', u.dimensionless_unscaled),
-                             ('300 - 800 keV', u.dimensionless_unscaled),
-                             ('800 - 7000 keV', u.dimensionless_unscaled),
-                             ('7000 - 20000 keV', u.dimensionless_unscaled)])
+        units = OrderedDict([('3 - 6 keV', u.ct / u.s / u.Unit('detector')),
```
Comment (Cadair) [inline sunpy/timeseries/sources/rhessi.py:128]: what is this detector unit? how is this going to fall out if people want to use this data for something else?

File: sunpy/timeseries/timeseriesbase.py [hunk 3]
```diff
@@ -26,6 +26,10 @@
 from astropy.table import Table
 from astropy.table import Column
 
+# define and register a new unit, needed for RHESSI
+det = u.def_unit('detector')
+u.add_enabled_units([det])
```
Comment (Cadair) [inline sunpy/timeseries/timeseriesbase.py:31]: what does `add_enabled_units` do?

File: sunpy/timeseries/timeseriesbase.py [hunk 4]
```diff
@@ -334,6 +335,10 @@ def concatenate(self, otherts, **kwargs):
         at once.
         """
 
+        # check to see if nothing needs to be done
+        if self == otherts:
```
Comment (Cadair) [inline sunpy/timeseries/timeseriesbase.py:339]: how is this going to be computed? is `__eq__` defined? would this bet better as `is`??

File: sunpy/timeseries/sources/rhessi.py (hunk 2 above)
Comment (ehsteve) [reply sunpy/timeseries/sources/rhessi.py:128]: This is a new unit that I am defining. It is used by RHESSI and would be useful generally. It is defined below.

File: sunpy/timeseries/timeseriesbase.py (hunk 3 above)
Comment (ehsteve) [reply sunpy/timeseries/timeseriesbase.py:31]: This adds a new unit to the unit registry which enables users to use it. It is only defined when they import timeseries. We could consider making a PR to astropy itself to add it.

File: sunpy/timeseries/timeseriesbase.py (hunk 4 above)
Comment (ehsteve) [reply sunpy/timeseries/timeseriesbase.py:339]: Yes, @Alex-Ian-Hamilton defined it earlier. It mostly just checks if the data is the same.

Comment (Alex-Ian-Hamilton) [reply sunpy/timeseries/timeseriesbase.py:339]: It checks that the data, units and metadata for the two TS match. If so they are considered to be equivalent.
In this case I used Pandas functionality to compare the data DataFrames and compared the list storing metadata (tuple) entries in the metadata. The units are simply a dictionary and this is compared directly.

File: sunpy/timeseries/sources/rhessi.py (hunk 2 above)
Comment (Cadair) [reply sunpy/timeseries/sources/rhessi.py:128]: I am not quite sure I understand the context of the unit, how is it per detector?!

File: sunpy/timeseries/timeseriesbase.py (hunk 3 above)
Comment (Cadair) [reply sunpy/timeseries/timeseriesbase.py:31]: so it adds it to the `astropy.units` namespace?

File: sunpy/timeseries/timeseriesbase.py (hunk 4 above)
Comment (Cadair) [reply sunpy/timeseries/timeseriesbase.py:339]: cool

File: sunpy/timeseries/sources/rhessi.py (hunk 2 above)
Comment (ehsteve) [reply sunpy/timeseries/sources/rhessi.py:128]: RHESSI has 9 detectors and they are frequently combined so it is necessary to say whether a value is per detector or for all detectors (and therefore close to 9 times larger).

File: sunpy/timeseries/timeseriesbase.py (hunk 3 above)
Comment (ehsteve) [reply sunpy/timeseries/timeseriesbase.py:31]: It enables Units('detector') but I don't think that u.detector works.

File: sunpy/timeseries/sources/rhessi.py (hunk 2 above)
Comment (Cadair) [reply sunpy/timeseries/sources/rhessi.py:128]: Ok, that makes sense, this needs documenting somewhere.

File: sunpy/timeseries/timeseriesbase.py (hunk 3 above)
Comment (Cadair) [reply sunpy/timeseries/timeseriesbase.py:31]: right ok cool.

File: sunpy/timeseries/sources/rhessi.py (hunk 2 above)
Comment (ehsteve) [reply sunpy/timeseries/sources/rhessi.py:128]: I updated the doc string to mention this fact.

Comment (Cadair) [reply sunpy/timeseries/sources/rhessi.py:128]: Cool, that's a good explanation, thanks. Is there also somewhere suitable (in the same place?) to mention the new unit, just so it doesn't catch people off guard.

**Pull request merged**